homeassistant/components/homekit/type_cameras.py

```diff
@@ -313,51 +313,41 @@ async def start_stream(self, session_info, stream_config):
         if not opened:
             _LOGGER.error("Failed to open ffmpeg stream")
             return False
-        session_info["stream"] = stream
+
         _LOGGER.info(
             "[%s] Started stream process - PID %d",
             session_info["id"],
             stream.process.pid,
         )
 
-        ffmpeg_watcher = async_track_time_interval(
-            self.hass, self._async_ffmpeg_watch, FFMPEG_WATCH_INTERVAL
+        session_info["stream"] = stream
+        session_info[FFMPEG_PID] = stream.process.pid
+        session_info[FFMPEG_WATCHER] = async_track_time_interval(
+            self.hass,
+            lambda _: self._async_ffmpeg_watch(session_info["id"]),
+            FFMPEG_WATCH_INTERVAL,
         )
-        self._cur_session = {
-            FFMPEG_WATCHER: ffmpeg_watcher,
-            FFMPEG_PID: stream.process.pid,
-            SESSION_ID: session_info["id"],
-        }
 
-        return await self._async_ffmpeg_watch(0)
+        return await self._async_ffmpeg_watch(session_info["id"])
 
-    async def _async_ffmpeg_watch(self, _):
+    async def _async_ffmpeg_watch(self, session_id):
         """Check to make sure ffmpeg is still running and cleanup if not."""
-        ffmpeg_pid = self._cur_session[FFMPEG_PID]
-        session_id = self._cur_session[SESSION_ID]
+        ffmpeg_pid = self.sessions[session_id][FFMPEG_PID]
         if pid_is_alive(ffmpeg_pid):
             return True
 
         _LOGGER.warning("Streaming process ended unexpectedly - PID %d", ffmpeg_pid)
-        self._async_stop_ffmpeg_watch()
-        self._async_set_streaming_available(session_id)
+        self._async_stop_ffmpeg_watch(session_id)
+        self.set_streaming_available(self.sessions[session_id]["stream_idx"])
         return False
 
     @callback
-    def _async_stop_ffmpeg_watch(self):
+    def _async_stop_ffmpeg_watch(self, session_id):
         """Cleanup a streaming session after stopping."""
-        if not self._cur_session:
+        if FFMPEG_WATCHER not in self.sessions[session_id]:
             return
-        self._cur_session[FFMPEG_WATCHER]()
-        self._cur_session = None
-
-    @callback
-    def _async_set_streaming_available(self, session_id):
-        """Free the session so they can start another."""
-        self.streaming_status = STREAMING_STATUS["AVAILABLE"]
-        self.get_service(SERV_CAMERA_RTP_STREAM_MANAGEMENT).get_characteristic(
-            CHAR_STREAMING_STRATUS
-        ).notify()
+        self.sessions[session_id][FFMPEG_WATCHER]()
+        del self.sessions[session_id][FFMPEG_WATCHER]
```

Comment:
Not perse better, just a possibility:

```suggestion
        self.sessions[session_id].pop(FFMPEG_WATCHER)()
```